sgt-back-end/server.js

```diff
@@ -0,0 +1,202 @@
+import express from 'express';
+import pg from 'pg';
+
+const app = express();
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+// ---- Reusable functions ----
+function handleError(res, err) {
+  console.error('/api/grades error', err);
+  res.status(500).json({ error: 'An unexpected error occurred.' });
+}
+
+function evalParamId(res, id) {
+  if (isNaN(id) || Number(id) < 0 || !Number.isInteger(id)) {
+    res.status(400).json({ error: "'gradeId' must be a positive integer." });
+  } else {
+    return true;
+  }
+}
+
+function evalReqBody(req, res) {
+  const userInputs = req.body;
+  const score = userInputs.score;
+  const inputKeys = Object.keys(userInputs);
+  const reqKeys = ['course', 'name', 'score'];
+  if (inputKeys.length < 3) {
+    res.status(400).json({
+      error: "Must include 'course', 'name', and 'score'. Please try again."
+    });
+    return false;
+  }
+  if (inputKeys.length > 3) {
+    res.status(400).json({
+      error: "Only 'course', 'name', and 'score' would be processed. Please try again."
+    });
+    return false;
+  }
+  for (const key of inputKeys) {
+    if (reqKeys.indexOf(key) === -1) {
+      res.status(400).json({
+        error: "Only 'course', 'name', and 'score' would be processed. Please try again."
+      });
+      return false;
+    }
+    if (!userInputs[key]) {
+      res.status(400).json({
+        error: `'${key}' was left blank. Please resubmit and include a value for '${key}'.`
+      });
+      return false;
+    }
+  }
+  if (isNaN(score) || Number(score) < 0) {
+    res.status(400).json({ error: `${score} is not a valid 'score'. Scores must be positive numbers.` });
+    return false;
+  } else {
```

Comment:
This `else` is redundant since the only way you can get here is if all the other tests pass.

Some people like having the redundant `else` there, but I personally don't. It's a matter of preference. However, none of your other `if` statements above have a redundant `else`. :-) Consistency is important.